libzecale/circuits/aggregator_gadget.tcc

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#ifndef __ZECALE_CIRCUITS_AGGREGATOR_GADGET_TCC__
+#define __ZECALE_CIRCUITS_AGGREGATOR_GADGET_TCC__
+
+#include "libzecale/circuits/aggregator_gadget.hpp"
+
+namespace libzecale
+{
+
+template<
+    typename nppT,
+    typename wppT,
+    typename nsnarkT,
+    typename wverifierT,
+    size_t NumProofs>
+aggregator_gadget<nppT, wppT, nsnarkT, wverifierT, NumProofs>::
+    aggregator_gadget(
+        libsnark::protoboard<libff::Fr<wppT>> &pb,
+        const size_t inputs_per_nested_proof,
+        const std::string &annotation_prefix)
+    : libsnark::gadget<libff::Fr<wppT>>(pb, annotation_prefix)
+    , num_inputs_per_nested_proof(inputs_per_nested_proof)
+{
+    // Assert that a single nested input can be encoded in a single wrapper
+    // input. This holds for all pairing chains we test, but if it fails then
+    // nested inputs will need to be encoded over multiple wrapper inputs.
```

Comment:
`wrapper inputs` -> scalar field elements (of the wrapping curve).

The "nested inputs" are assigned to wires in the arithmetic circuit which itself is defined over $\FF_{r_{BW6}}$.

Reply:
I've reworded to be more explicit about the fields.  Not sure if I've fully understood the issue, so let me know if the new version does address it.

Reply:
This is just a formulation thing. The Zecale statement is represented as an arithmetic circuit defined over the scalar field of BW6 (i.e. the wrapping curve). Hence, I was suggesting

> [...] but if it fails, then nested inputs will need to be encoded over multiple scalar field elements of the wrapping curve [...]

Reply:
You rephrasing is clear, so let's keep it that way